### ISSUE TITLE
Warning  debug_toolbar.W006

### ISSUE THREAD

**Issue by rjdini** [tendenci/tendenci#1099]
I receive the following warning in response to python manage.py check on  rev 14.0.4

Any suggestions, please??

WARNINGS:
tendenci_1  | ?: (debug_toolbar.W006) At least one DjangoTemplates TEMPLATES configuration needs to have APP_DIRS set to True.
tendenci_1  | 	HINT: Use APP_DIRS=True for at least one django.template.backends.django.DjangoTemplates backend configuration.

**Comment by kbaichtal** [comment]
I am having this issue too.

**Comment by rjdini** [comment]
I also continue to have this error (now on rev 14.1).  No suggestions so far....

**Comment by jennyq** [comment]
Hi @rjdini,

It appears it was an issue in django-debug-toolbar. The warning message is gone after django-debug-toolbar is updated to the latest version 3.4.0. 

Below is their updated debug_toolbar.W006 warning:
debug_toolbar.W006: At least one DjangoTemplates TEMPLATES configuration needs to have django.template.loaders.app_directories.Loader included in ["OPTIONS"]["loaders"] or APP_DIRS set to True. 

As you can see, tendenci does have "django.template.loaders.app_directories.Loader" in  ["OPTIONS"]["loaders"] https://github.com/tendenci/tendenci/blob/master/tendenci/settings.py#L132

**Comment by jennyq** [comment]
Hi @kbaichtal,

The conf/settings.py in your project is different from the settings.py in tendenci core.  Please don't remove it. https://github.com/tendenci/tendenci-project-template/blob/master/conf/settings.py#L1

The conf/settings.py in your project imports everything from the settings.py, and from there, you can update/override some of the settings to fit your needs.

**Comment by kbaichtal** [comment]
I already had django-debug-toolbar v3.4.0, and my installation doesn't work. Basically nothing happens when I go to my site url. I'm pretty sure this error is the root of it because everything I've tried comes back to it.

(venv_aba_ams) ***@***.***:/var/www/www_ashlandbusiness_com_secure/aba_ams$ pip install django-debug-toolbar --upgrade
Requirement already satisfied: django-debug-toolbar in /home/aba/venv_aba_ams/lib/python3.9/site-packages (3.4.0)


(venv_aba_ams) ***@***.***:/var/www/www_ashlandbusiness_com_secure/aba_ams$ python manage.py update_dashboard_stats
System check identified some issues:
WARNINGS:
?: (debug_toolbar.W006) At least one DjangoTemplates TEMPLATES configuration needs to use django.template.loaders.app_directories.Loader or have APP_DIRS set to True.
        HINT: Include django.template.loaders.app_directories.Loader in ["OPTIONS"]["loaders"]. Alternatively use APP_DIRS=True for at least one django.template.backends.django.DjangoTemplates backend configuration.
Creating dashboard statistics for upcoming events
Creating dashboard statistics for form submissions
Creating dashboard statistics for pages traffic 2022-06-03 11:20:58.183115
Creating dashboard statistics for events traffic 2022-06-03 11:20:58.189441
Creating dashboard statistics for memberships 2022-06-03 11:20:58.195037
Creating dashboard statistics for new memberships
Creating dashboard statistics for renewed memberships
Creating dashboard statistics for expired memberships
Creating dashboard statistics for expiring memberships
Creating dashboard statistics for new corporate memberships
Creating dashboard statistics for renewed corporate memberships
Creating dashboard statistics for expired corporate memberships
Creating dashboard statistics for expiring corporate memberships
Creating dashboard statistics for top corporate members

I really appreciate the advice you've been giving, and hope you can help me get to the bottom of this. Thank you!


-----Original Message-----

**Comment by jennyq** [comment]
@kbaichtal, 

It's just a warning. It should not affect the installation. That warning will be gone after a new release is pushed to PYPI. For now, you can ignore the warnings with the "-W" option, for example, `python -W manage.py migrate`.

What errors are you getting besides that?

**Comment by kbaichtal** [comment]
The web server just says file not found instead of producing the tendenci website. Logs say:

[2022-06-03 13:11:29.961081] [socache_shmcb:debug] - - AH00831: socache_shmcb_store (0x2d -> subcache 13)
[2022-06-03 13:11:29.961106] [socache_shmcb:debug] - - AH00847: insert happened at idx=0, data=(0:32)
[2022-06-03 13:11:29.961110] [socache_shmcb:debug] - - AH00848: finished insert, subcache: idx_pos/idx_used=0/1, data_pos/data_used=0/228
[2022-06-03 13:11:29.961113] [socache_shmcb:debug] - - AH00834: leaving socache_shmcb_store successfully
[2022-06-03 13:11:29.961173] [socache_shmcb:debug] - - AH00831: socache_shmcb_store (0x7c -> subcache 28)
[2022-06-03 13:11:29.961190] [socache_shmcb:debug] - - AH00847: insert happened at idx=0, data=(0:32)
[2022-06-03 13:11:29.961193] [socache_shmcb:debug] - - AH00848: finished insert, subcache: idx_pos/idx_used=0/1, data_pos/data_used=0/228
[2022-06-03 13:11:29.961203] [socache_shmcb:debug] - - AH00834: leaving socache_shmcb_store successfully
[2022-06-03 13:11:29.961299] [ssl:debug] 192.168.1.130:49398 M0EzDf8NSAI AH02034: Initial (No.1) HTTPS request received for child 128 (server www.ashlandbusiness.com:443)
[2022-06-03 13:11:29.961346] [authz_core:debug] 192.168.1.130:49398 M0EzDf8NSAI AH01626: authorization result of Require all granted: granted
[2022-06-03 13:11:29.961359] [authz_core:debug] 192.168.1.130:49398 M0EzDf8NSAI AH01626: authorization result of <RequireAny>: granted
[2022-06-03 13:11:29.961391] [authz_core:debug] 192.168.1.130:49398 M0EzDf8NSAI AH01626: authorization result of Require all granted: granted
[2022-06-03 13:11:29.961396] [authz_core:debug] 192.168.1.130:49398 M0EzDf8NSAI AH01626: authorization result of <RequireAny>: granted
[2022-06-03 13:11:29.961417] [authz_core:debug] 192.168.1.130:49398 M0EzDf8NSAI AH01626: authorization result of Require all granted: granted
[2022-06-03 13:11:29.961422] [authz_core:debug] 192.168.1.130:49398 M0EzDf8NSAI AH01626: authorization result of <RequireAny>: granted
[2022-06-03 13:11:29.961437] [core:info] 192.168.1.130:49398 M0EzDf8NSAI AH00129: Attempt to serve directory: /var/www/www_ashlandbusiness_com_secure/
[2022-06-03 13:11:30.144785] [ssl:debug] 192.168.1.130:49398 1g02DT8NSAI AH02034: Subsequent (No.2) HTTPS request received for child 129 (server www.ashlandbusiness.com:443)
[2022-06-03 13:11:30.144901] [authz_core:debug] 192.168.1.130:49398 1g02DT8NSAI AH01626: authorization result of Require all granted: granted
[2022-06-03 13:11:30.144920] [authz_core:debug] 192.168.1.130:49398 1g02DT8NSAI AH01626: authorization result of <RequireAny>: granted
[2022-06-03 13:11:30.144950] [core:info] 192.168.1.130:49398 1g02DT8NSAI AH00128: File does not exist: /var/www/www_ashlandbusiness_com_secure/aba_ams/favicon.ico
[2022-06-03 13:11:35.145942] [ssl:debug] 192.168.1.130:49398 - AH02001: Connection closed to child 130 with standard shutdown (server www.ashlandbusiness.com:443)
[2022-06-03 13:11:50.712826] [socache_shmcb:debug] - - AH00831: socache_shmcb_store (0x65 -> subcache 5)
[2022-06-03 13:11:50.712894] [socache_shmcb:debug] - - AH00847: insert happened at idx=0, data=(0:32)
[2022-06-03 13:11:50.712905] [socache_shmcb:debug] - - AH00848: finished insert, subcache: idx_pos/idx_used=0/1, data_pos/data_used=0/228
[2022-06-03 13:11:50.712917] [socache_shmcb:debug] - - AH00834: leaving socache_shmcb_store successfully
[2022-06-03 13:11:55.731710] [ssl:debug] 192.168.1.130:49401 - AH02001: Connection closed to child 0 with standard shutdown (server www.ashlandbusiness.com:443)

The only other error I can find is if I do check_settings, two more are added under the W006 one:

(venv_aba_ams) ***@***.***:/var/www/www_ashlandbusiness_com_secure/aba_ams$ python manage.py check_settings
System check identified some issues:

WARNINGS:
?: (debug_toolbar.W006) At least one DjangoTemplates TEMPLATES configuration needs to use django.template.loaders.app_directories.Loader or have APP_DIRS set to True.
        HINT: Include django.template.loaders.app_directories.Loader in ["OPTIONS"]["loaders"]. Alternatively use APP_DIRS=True for at least one django.template.backends.django.DjangoTemplates backend configuration.
Setting: corporatemembershiprecipients module corporatememberships, is missing.
Setting: servicerecipients module service, is missing.


Thank you.

-----Original Message-----

**Comment by jennyq** [comment]
Have you run the command `python manage.py load_tendenci_defaults`?
https://tendenci.readthedocs.io/en/latest/installation/installation.html#setting-up-the-site

**Comment by kbaichtal** [comment]
Yes, I have done all of those multiple times. It's all good, just no website appears. And those two errors on check_settings.

**Comment by jennyq** [comment]
Can you run `python manage.py runserver 8002`, then in another terminal, run `curl http://localhost:8002/` to see what you get?

**Comment by kbaichtal** [comment]
That does work.

**Comment by jennyq** [comment]
Have you set up a systemd service for the site? If so, check the status `sudo systemctl status mysite`.
https://tendenci.readthedocs.io/en/latest/installation/installation.html#systemd-setup

**Comment by kbaichtal** [comment]
With your help, I finally figured it out! My machine does not have systemd (it is a Slackware box) but I determined that the project was in fact running. BUT, I had remmed out one of the two nearly identically-named VirtualServers using the wsgi process, to reduce the complexity while hammering out installation issues. It turns out I was attempting to access the website via the other VirtualServer's name. Boy do I feel silly now. Once I corrected that detail of my test procedure, I was able to tweak everything to work from there.

Thank you for your patience, I really appreciate it.


-----Original Message-----
